src/Go/boardAnalysis/goAI.ts

```diff
@@ -15,20 +15,29 @@ import {
   getAllEyesByChainId,
   getAllNeighboringChains,
   getAllValidMoves,
+  getPreviousMoveDetails,
 } from "./boardAnalysis";
 import { findDisputedTerritory } from "./controlledTerritory";
 import { findAnyMatchedPatterns } from "./patternMatching";
 import { WHRNG } from "../../Casino/RNG";
 import { Go, GoEvents } from "../Go";
 
 let currentAITurn: Promise<Play> | null = null;
+let currentTurnResolver: ((value: Play | PromiseLike<Play>) => void) | null = null;
 
 /**
  * Retrieves a move from the current faction in response to the player's move
  */
 export function makeAIMove(boardState: BoardState): Promise<Play> {
   // If AI is already taking their turn, return the existing turn.
-  if (currentAITurn) return currentAITurn;
+  if (currentAITurn) {
+    return Go.nextTurn;
```

Comment:
This is a subtle change. Remind me why you're doing it?

Reply:
The linter should do it for us, but it missed this one it seems. It should be a ternary (or short-circuit &&) for one line conditionals, for all of the standard maintainability reasons

https://www.reddit.com/r/C_Programming/comments/iyu0ei/an_argument_on_why_you_should_always_put_curly/

Reply:
I wasn't clear enough. I'm fully onboard with adding curly braces; the subtle change is the switch from `currentAITurn` to `Go.nextTurn`.

Reply:
Ah! My first plan was to remove currentAITurn entirely, then it became a boolean flag later, and the above code is the result of that being only partway changed. It makes more sense with the most recent version of the code using `isAiThinking` as the boolean conditional.

Reply:
Ok, that's all well and good, but it doesn't answer my original question :D

Changing from `currentAITurn` to `Go.nextTurn` here is a subtle change, with subtle implications. They were *almost* the same, but not quite in all circumstances. I can be pretty sure I know what's going on with `currentAITurn`, since it's scoped to this file, but `Go.nextTurn` is effectively global and could be modified anywhere. 

So, why change it?

Reply:
Go.nextTurn is the source of truth for the current turn. In the best case the stuff in this local file stays up-to-date, but (as I discovered during implementing this) this file is not the only place where the white player could start or stop thinking, and having multiple locations editing multiple values that are supposed to be in sync gets messy. If currentAITurn  is truthy it should always be the same as Go.nextTurn, and if it wasn't then it is out-of-date and Go.nextTurn represents the current move's promise.

currentAITurn was added (by snarling, IIRC) because JS doesn't have a way to detect on a promise object if it is resolved or not. I considered making my own promise object type, or always await Go.nextTurn() and then checked on the game state in this file, but both seemed clunky to me